chips/fe310-g002/qemu.sh

```diff
@@ -1 +1 @@
-qemu-system-riscv32 -M sifive_u,msel=1 -nographic -serial mon:stdio -device loader,addr=0x20010000,cpu-num=0,file=final.bin -semihosting -semihosting-config enable=on,userspace=on
+qemu-system-riscv32 -M sifive_u,msel=1 -nographic -serial mon:stdio -device loader,addr=0x20010000,cpu-num=0,file=final.ihex -semihosting -semihosting-config enable=on,userspace=on
```

Comment:
Unable to post a comment on the commit so posting it here. Why does qemu allow `.ihex` but not '.bin`? Does it mean that the image is being loaded over something that is not RAM? Is it because this is being fetched from flash?

Reply:
It does allow `.bin`. However, when checking if the image is within the allowable size, the max size for a `.bin` is the size of the ram (in fact it only does this check for binaries).  See the[ qemu generic loader](https://github.com/qemu/qemu/blob/b3cd3b5a66f0dddfe3d5ba2bef13cd4f5b89cde9/hw/core/generic-loader.c#L158).  I think ihex contains addresses, so qemu can extract the size, but it cannot do so with a raw binary(?)

Reply:
I think `.bin` is an ELF so it should have the required information about where different components end up in RAM. I am just worried if using `.ihex` is working around some incorrect address loading being caught by the `.bin`/elf path. Did you trace the elf path to see what information it is looking for and based on what it decides that the required regions of the binary cannot fit in RAM?

Reply:
I did trace the path for the `.bin` but it just goes straight to the size check, even before trying to parse the data.  
While on the topic though, I do have an elf bug I still need to hunt down.  There seems to be an address issue with the final elf.  When trying to use the final.elf with qemu, it is getting incorrectly loaded.

Reply:
> I did trace the path for the .bin but it just goes straight to the size check, even before trying to parse the data.

I think we should make the loading of elf and ihex both work. Can you please raise an issue for this? We can debug that separately. I don't think we need to block this PR for that.

> While on the topic though, I do have an elf bug I still need to hunt down. There seems to be an address issue with the final elf. When trying to use the final.elf with qemu, it is getting incorrectly loaded.

Interesting. Let's raise bugs for each of these issues and track them.

Reply:
The `.elf` should be fixable, but I think loading the `.bin` would require a qemu patch.  As long as the `.bin` is smaller than the ram size, it loads and runs just fine.

https://rivosinc.atlassian.net/browse/SW-555
and 
https://rivosinc.atlassian.net/browse/SW-554

I am not sure if the `.bin` problem belongs in the hubris epic, but we can always move it.